go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/caddyserver/caddy/v2
 
-go 1.20
+go 1.21
```

Comment:
I don't think we're ready to bump our minimum version yet. Is this necessary?

Reply:
quic-go dropped support Go 1.20 a bit earlier than usual this team. See release notes for justification.

So yes, updating to v0.41.0 will unfortunately force you move away from Go 1.20.

Reply:
Okay. It might take us a little while to merge this. I think our next release v2.8.0 is still a little ways away (partly cause we have a lot of pending PRs to review and merge, partly because Matt is busy being a parent). So I think by the time we'll be ready, 1.22 will be out.

Reply:
> I think our next release v2.8.0 is still a little ways away

A smaller patch release with #5982 would still be good though.

Reply:
Understood @bt90 but that would take time to make a branch & cherrypick etc. I don't think @mholt has the time to do that without it being sponsored/paid for.

Reply:
Go 1.22 is slated for just a couple/few weeks. I'm willing to bet v2.8 won't be ready before then. Should we just merge this?

Reply:
@mholt sure, we'll need to also update CI config (and repo config) to test 1.21 + 1.22

Reply:
The justification for dropping Go 1.20 support is really good: no more crypto/tls fork.

One other thing to consider is that v0.42 will have an improved API for a security feature. https://github.com/quic-go/quic-go/issues/3549#issuecomment-1905301886 -- we could also just wait until that gets released and kill 2 birds with 1 stone. Either way is fine with me.